protocol/protocol.go

```diff
@@ -126,6 +127,34 @@ func New(config ProtocolConfig) *Protocol {
 	return p
 }
 
+// CurrentState returns the current protocol state
+func (p *Protocol) CurrentState() State {
+	stateChan := make(chan State, 1)
+	p.stateTransitionChan <- protocolStateTransition{
+		stateRespChan: stateChan,
+	}
+	return <-stateChan
+}
+
+// IsDone checks if the protocol is in a done/completed state
+func (p *Protocol) IsDone() bool {
+	currentState := p.CurrentState()
+	if entry, exists := p.config.StateMap[currentState]; exists {
+		return entry.Agency == AgencyNone
+	}
```

Comment:
This should also return `true` if the current state is the same as the configured initial protocol state.